account_wallet/wizards/wizard_account_move_refund_wallet.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2022 ACSONE SA/NV (http://www.acsone.eu)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import _, fields, models
+
+
+class AccountMoveRefund(models.TransientModel):
+
+    _name = "wizard.account_move_refund.wallet"
+    _description = "Refund By Wallet"
+
+    account_wallet_type_id = fields.Many2one(
+        comodel_name="account.wallet.type",
+        string="Wallet type",
+        required=True,
+        ondelete="cascade",
+    )
+
+    amount = fields.Float(required=True, ondelete="cascade")
+
+    partner_id = fields.Many2one(
+        comodel_name="res.partner", string="Customer", required=True, ondelete="cascade"
+    )
+
+    invoice_date = fields.Date(string="Invoice Date", default=fields.Date.today)
+
+    product_id = fields.Many2one(
+        comodel_name="product.product",
+        string="Product",
+        required=True,
+        ondelete="cascade",
+    )
+
+    def apply(self):
+        self.ensure_one()
+        old_account = self.partner_id.property_account_receivable_id
+        self.partner_id.property_account_receivable_id = (
+            self.account_wallet_type_id.account_id
+        )
```

Comment:
This might raise permission issue if the user can make invoices but can't change the partner records (which is common in many organizations). If we don't fix this now, this should be mentioned in the roadmap section of the readme.

Reply:
i can tackle this now <br/>
i do the change via a sudo ? <br/>
I check before and raise my own exception ? <br/>

Reply:
IMHO this should be done with a context manager to ensure the operation will be rollbacked at the end of the apply.

It can be done through sudo() yes.

Reply:
I would say:

```suggestion
        with self.env.norecompute():
            self.partner_id.property_account_receivable_id = (
                self.account_wallet_type_id.account_id
            )
            (...)
            self.partner_id.property_account_receivable_id = old_account
```

@sbidoul What do you think ?

Reply:
And better :

```suggestion
        with self.env.norecompute():
            try:
                    self.partner_id.property_account_receivable_id = (
                        self.account_wallet_type_id.account_id
                    )
                    (...)
           finally:
                   self.partner_id.property_account_receivable_id = old_account
```

Reply:
> IMHO this should be done with a context manager to ensure the operation will be rollbacked at the end of the apply.

Sounds overkill. A try/finally maybe ? In any case if the creation fails, there will be a rollback of the transaction anyway.

About the norecompute I don't know. Do you have a specific issue in mind that it would resolve ?

Any any case, it's a hack but if there is no better way, so be it, we need to move forward.

Reply:
Ok, try/finally is maybe sufficent